app/medInria/medDataSourceManager.cpp

```diff
@@ -178,6 +185,80 @@ void medDataSourceManager::loadFromPath(QString path)
     medDataManager::instance()->importPath(path, false);
 }
 
+void medDataSourceManager::fuseFromPaths(QStringList paths)
+{
+    d->indexList.clear();
+    QList<QUuid> uuidList;
+    d->numberOfPaths = paths.count();
+
+    connect(medDataManager::instance(), SIGNAL(dataImported(medDataIndex,QUuid)),
+            this, SLOT(dataImported(medDataIndex,QUuid)));
+
+    // Non persistent (temporary) loading
+    foreach(QString path, paths)
+    {
+        uuidList.append(medDataManager::instance()->importPath(path, false));
+    }
+
+    // Wait for the end of loadings
+    QEventLoop loop;
+    connect(this, SIGNAL(loadingsFinished()), &loop, SLOT(quit()));
+    loop.exec();
+
+    disconnect(medDataManager::instance(), SIGNAL(dataImported(medDataIndex,QUuid)),
+               this, SLOT(dataImported(medDataIndex,QUuid)));
```

Comment:
I would rename the SLOT, because ```dataImported``` is more a SIGNAL name.
I am not really inspired because this SLOT doesn't do much, so maybe ```onDataImported``` can be fine (although I man not a big fan of this way of naming).